### ISSUE TITLE
bazel doesn't support sh_binary with pl file.

### ISSUE THREAD

**Issue by damienmg** [bazelbuild/bazel#1468]
More specifically, using a pl file as a tool from a skylark rule will not work because of src/main/java/com/google/devtools/build/lib/util/CommandBuilder.java#L49

We should allow more runner.
Internal ref: b/29485606

**Comment by forumulator** [comment]
@damienmg Can you please elaborate a little more on this? What exactly do you mean by "using a pl file as a tool from skylark rule"? I'm trying to work on this.

**Comment by damienmg** [comment]
The problem is that we cannot use  Perl script as an executable input for
Skylark because we restrict the file types that can be executable when they
are not coming from rules.

**Comment by PiotrSikora** [comment]
FWIW, there is [bazelbuild/rules_perl](https://github.com/bazelbuild/rules_perl).

**Comment by damienmg** [comment]
Sorry so I went again on the bug. The basic problem is perl files can be executable but are not allowed as source for sh_binary. As pointed by @PiotrSikora, bazelbuild/rules_perl allow people to depends on perl binary but ultimately there is no reason for being so restrictive in sh_binary

**Comment by forumulator** [comment]
But the sh_binary does accept .pl files in it's source list. For example, the following rule builds and runs as expected 
``
sh_binary (
    name = "pl_rule",
    srcs = ["a.pl"]
)
``
I still didn't grasp the exact problem. (I'm sorry about that, I'm kinda new here).

**Comment by royalharsh** [comment]
I couldn't reproduce this bug. Here's my BUILD file:
```
sh_binary(
  name = "pl",
  srcs = ["hello.pl"],
)
```
I also tried with a perl executable(made with [pp](http://search.cpan.org/~autrijus/PAR-0.85_01/script/pp)): 
```
sh_binary(
  name = "pl",
  srcs = ["hello"],
)
```
But everything seems to work fine.
```bash
$ ~/bazel/bazel-bin/src/bazel build -s :pl     
INFO: Found 1 target...
>>>>> # //:pl [action 'Creating runfiles tree bazel-out/local-fastbuild/bin/pl.runfiles']
(cd /home/hvardhan/.cache/bazel/_bazel_hvardhan/0e635913dc4fb37eccaf9a0d84aa2977/execroot/test_bazel && \
  exec env - \
  /home/hvardhan/.cache/bazel/_bazel_hvardhan/0e635913dc4fb37eccaf9a0d84aa2977/execroot/test_bazel/_bin/build-runfiles bazel-out/local-fastbuild/bin/pl.runfiles_manifest bazel-out/local-fastbuild/bin/pl.runfiles)
Target //:pl up-to-date:
  bazel-bin/pl
INFO: Elapsed time: 0.265s, Critical Path: 0.04s
```
It would be great if you could explain a bit on how you ran into this bug?

**Comment by damienmg** [comment]
Ok I digged into the history and that bug has been fixed long ago by @jmmv (commit 0c0bfe6297bdc6c061a88c91c4eaff1776f6ef23)
